app/models/bidding.rb

```diff
@@ -0,0 +1,5 @@
+class Bidding < ApplicationRecord
+  validates :bid, presence: true, numericality: { only_integer: true }
+  belongs_to :listing
+  belongs_to :user
```

Comment:
Add a `before_create` hook here where you check if the current user that is submitting the bid is the same as the landlord of the listing that he wants to bid on. If that is the case raise an error.

Reply:
Also add a test for this